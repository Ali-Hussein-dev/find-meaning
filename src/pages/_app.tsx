import { ChakraProvider } from '@chakra-ui/react';
import '../css/global.css';
import { extendTheme } from '@chakra-ui/react';
if (process.env.NEXT_PUBLIC_API_MOCKING === 'enabled') {
  import('@/test/__mocks__/index');
}

const theme = extendTheme({
  styles: {
    global: {
      body: {
        fontFamily: 'Poppins, Sans-serif',
      },
    },
  },
});
const MyApp: React.FC<{ Component: React.FC; pageProps: any }> = ({
  Component,
  pageProps,
}): JSX.Element => {
  return (
    <ChakraProvider theme={theme}>
        <link rel="icon" href="/favicon.ico" />
      <div className="bg">
        <Component {...pageProps} />
      </div>
    </ChakraProvider>
  );
};

export default MyApp;
