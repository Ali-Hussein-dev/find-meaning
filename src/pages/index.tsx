import { SearchBar } from '@/components/index';
import Head from 'next/head';
import * as React from 'react';
import { isMobile } from 'react-device-detect';
import Fade from 'react-reveal/Fade';
import Zoom from 'react-reveal/Zoom';

//--------------------------------------
export const Home = (): JSX.Element => {
  switch (isMobile) {
    case true:
      {
        //--------------------------------------hooks
        const [mobSearchfocused, setMobSearchfocused] = React.useState(false);
        React.useEffect(() => {
          if (document && mobSearchfocused) {
            //@ts-ignore
            document.body.querySelector('#bottom input')?.focus();
          }
        }, [mobSearchfocused]);
        return (
          <>
            <Head>
              <title>Find Meaning</title>
            </Head>
            <main
              className="relative grid w-full h-[93vh] place-items-center bg-blueGray-800"
              onMouseDown={(e) => {
                e.stopPropagation();
                setMobSearchfocused(false);
              }}
            >
              <div className="w-full sm:w-[640px] px-4">
                <Zoom>
                  <img
                    src="/icons/fm_logo.svg"
                    className="w-32 h-32 mx-auto"
                    alt="logo"
                  />
                </Zoom>
                <div className="center">
                  <h1 className="mb-2 text-2xl font-semibold text-center text-orange-700">
                    Find Meaning
                  </h1>
                  <span className="ml-1 text-xs text-blueGray-400">beta</span>
                </div>
                <div
                  onMouseDown={(e) => {
                    e.stopPropagation();
                    setMobSearchfocused(true);
                  }}
                  hidden={mobSearchfocused}
                >
                  <Fade bottom wait={100}>
                    <SearchBar pushRouter={true} />
                  </Fade>
                </div>
              </div>
              <div
                id="bottom"
                hidden={!mobSearchfocused}
                className="fixed bottom-0 z-10 w-full pt-1 bg-blueGray-800"
                onMouseDown={(e) => {
                  e.stopPropagation();
                }}
              >
                <Fade bottom when={mobSearchfocused}>
                  <SearchBar pushRouter={true} />
                </Fade>
              </div>
            </main>
          </>
        );
      }
      break;

    default:
      return (
        <>
          <Head>
            <title>Find Meaning</title>
            <link rel="icon" href="/favicon/favicon.ico" />
          </Head>
          <main className="relative grid w-full h-[93vh] place-items-center bg-blueGray-800">
            <div className="w-full sm:w-[640px] px-4 mb-48">
              <Zoom>
                <img
                  src="/icons/fm_logo.svg"
                  className="w-32 h-32 mx-auto"
                  alt="logo"
                />
              </Zoom>
              <div className="center">
                <h1 className="mb-2 text-2xl font-semibold text-center text-orange-700">
                  Find Meaning
                </h1>
                <span className="ml-1 text-xs text-blueGray-400">beta</span>
              </div>
              <div></div>
              <Fade bottom wait={100}>
                <SearchBar pushRouter={true} />
              </Fade>
            </div>
          </main>
        </>
      );
      break;
  }
};

export default Home;
