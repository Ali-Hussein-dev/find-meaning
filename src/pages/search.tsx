import { SearchBar, WikiResponse } from '@/components/index';
import * as React from 'react';
import { QueryClient, QueryClientProvider } from 'react-query';
import { ErrorBoundary } from 'react-error-boundary';
import dynamic from 'next/dynamic';
const BgParticleDynamic = dynamic(() =>
  import('../../src/components/index').then((com) => com.BgParticles),
);
//--------------------------------------
interface CtxFreshData {
  freshData: any;
  setFreshData: (prvState: any) => void;
}

const queryClient = new QueryClient();
export const Ctx = React.createContext<CtxFreshData | null>(null);
const Fallback = () => <div role="alert">Something went wrong!</div>;
//=======================
const SearchPage: React.FC = () => {
  // hooks
  const [freshData, setFreshData] = React.useState<any>(null);
  //--------------------------------------
  // functions
  //--------------------------------------
  return (
    <QueryClientProvider client={queryClient}>
      <Ctx.Provider value={{ freshData, setFreshData }}>
        <title>Find-Meaning | search</title>
        <main className="flex flex-col items-center min-h-screen px-1 bg-gradient-blueGray">
          <div
            data-testid="search-page"
            className="z-10 w-full px-1 sm:w-11/12 md:w-9/12 lg:w-8/12"
          >
            <SearchBar />
            <ErrorBoundary FallbackComponent={Fallback}>
              <WikiResponse />
            </ErrorBoundary>
          </div>
          <div className="fixed inset-0 z-0 h-screen">
            <BgParticleDynamic velocity={0.1} count={85} />
          </div>
        </main>
      </Ctx.Provider>
    </QueryClientProvider>
  );
};
export default SearchPage;
