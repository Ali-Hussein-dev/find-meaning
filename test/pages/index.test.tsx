import React from 'react';
import { render } from '../testUtils';
import { Home } from '../../src/pages/index';

describe('Home page', () => {
  it('Home page', () => {
    const { container, asFragment } = render(<Home />);
    expect(container).toBeInTheDocument();
    expect(asFragment()).toMatchSnapshot();
  });
});
