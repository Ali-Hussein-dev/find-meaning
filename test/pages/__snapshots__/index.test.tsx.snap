// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home page Home page 1`] = `
<DocumentFragment>
  <main
    class="relative grid w-full h-[93vh] place-items-center bg-blueGray-800"
  >
    <div
      class="w-full sm:w-[640px] px-4"
    >
      <div
        class="center"
      >
        <h1
          class="mb-2 text-2xl font-semibold text-center text-orange-700"
        >
          Find Meaning
        </h1>
        <span
          class="ml-1 text-xs text-blueGray-400"
        >
          beta
        </span>
      </div>
      <div />
      <div
        class="react-reveal"
      >
        <div>
          <div
            aria-expanded="false"
            aria-haspopup="listbox"
            aria-labelledby="downshift-0-label"
            class="relative flex flex-col w-full"
            role="combobox"
          >
            <form
              action="handlers.ts"
              class="relative flex items-center justify-between w-full focus-within:bg-blueGray-700 focus-within:opacity-90 border-lightBlue-400 focus-within:border-lightBlue-400 focus-within:border-b-2 border-b"
            >
              <label
                class="w-full"
                for="downshift-0-input"
                id="downshift-0-label"
              >
                <label
                  for="search-input"
                />
                <input
                  accesskey="j"
                  aria-autocomplete="list"
                  aria-label="search-input"
                  aria-labelledby="downshift-0-label"
                  autocomplete="off"
                  class="w-full h-12 py-1 text-lg disabled:text-trueGray-400 disabled:cursor-not-allowed focus:outline-none pl-3 bg-blueGray-800 text-blueGray-200"
                  cn="pl-3 bg-blueGray-800 text-blueGray-200"
                  id="downshift-0-input"
                  lang="en"
                  name="search-input"
                  placeholder="start looking up words..."
                  type="text"
                  value=""
                />
              </label>
              <div
                class="flex items-center text-trueGray-500 bg-blueGray-800"
              >
                <div
                  class="grid w-12 h-12 place-items-center"
                  data-testid="cc-container"
                >
                  <button
                    aria-label="submit"
                    class="icon-btn w-12 h-12 rounded-none text-blueGray-200"
                    disabled=""
                    role="button"
                    type="submit"
                  >
                    <svg
                      class="mx-auto"
                      fill="currentColor"
                      height="1em"
                      stroke="currentColor"
                      stroke-width="0"
                      viewBox="0 0 16 16"
                      width="1em"
                      xmlns="http://www.w3.org/2000/svg"
                    >
                      <path
                        clip-rule="evenodd"
                        d="M10.442 10.442a1 1 0 011.415 0l3.85 3.85a1 1 0 01-1.414 1.415l-3.85-3.85a1 1 0 010-1.415z"
                        fill-rule="evenodd"
                      />
                      <path
                        clip-rule="evenodd"
                        d="M6.5 12a5.5 5.5 0 100-11 5.5 5.5 0 000 11zM13 6.5a6.5 6.5 0 11-13 0 6.5 6.5 0 0113 0z"
                        fill-rule="evenodd"
                      />
                    </svg>
                  </button>
                   
                </div>
              </div>
            </form>
            <ul
              aria-labelledby="downshift-0-label"
              class="absolute z-30 w-full py-2 text-lg tracking-wide rounded-b-lg shadow-lg b bg-blueGray-800 top-12 text-lightBlue-300"
              hidden=""
              id="downshift-0-menu"
              role="listbox"
            />
          </div>
        </div>
      </div>
    </div>
  </main>
</DocumentFragment>
`;
